tiledb/api/src/datatype/strategy.rs

```diff
@@ -64,24 +62,76 @@ pub fn prop_datatype_implemented() -> impl Strategy<Value = Datatype> {
         Just(Datatype::UInt16),
         Just(Datatype::UInt32),
         Just(Datatype::UInt64),
-        Just(Datatype::Float32),
-        Just(Datatype::Float64),
+        Just(Datatype::DateTimeYear),
+        Just(Datatype::DateTimeMonth),
+        Just(Datatype::DateTimeWeek),
+        Just(Datatype::DateTimeDay),
+        Just(Datatype::DateTimeHour),
+        Just(Datatype::DateTimeMinute),
+        Just(Datatype::DateTimeSecond),
+        Just(Datatype::DateTimeMillisecond),
+        Just(Datatype::DateTimeMicrosecond),
+        Just(Datatype::DateTimeNanosecond),
+        Just(Datatype::DateTimePicosecond),
+        Just(Datatype::DateTimeFemtosecond),
+        Just(Datatype::DateTimeAttosecond),
+        Just(Datatype::TimeHour),
+        Just(Datatype::TimeMinute),
+        Just(Datatype::TimeSecond),
+        Just(Datatype::TimeMillisecond),
+        Just(Datatype::TimeMicrosecond),
+        Just(Datatype::TimeNanosecond),
+        Just(Datatype::TimePicosecond),
+        Just(Datatype::TimeFemtosecond),
+        Just(Datatype::TimeAttosecond),
```

Comment:
You're missing `Datatype::Bool` here. Its weird but that's core's definition.

Reply:
This was a good note to add a check that everything outside of this set cannot be used for a dense dimension.

Reply:
Updated the test in a way that you will hopefully like - we construct a schema and check that it succeeds only if we expected it to, and if it failed we make sure it's because of a datatype error.

Reply:
I do indeed like that! The only reason I hadn't discovered that is_byte_type disparity is because I hadn't gotten to testing dimensions directly yet.